lib/email-client.js

```diff
@@ -42,22 +42,29 @@ export default class EmailClient {
 	 * It's possible to pass a function to validate received emails. For example when you waiting for specific email - validator may check if expected email is present
 	 * @param {string} emailAddress - Email address from where to get emails
 	 * @param {function} validator - Optional function to validate received emails
-	 * @returns {Promise} - Resolved when there is no errors and validator returns truthy
+	 * @returns {Object} - Returns `object`
 	 */
-	async pollEmailsByRecipient( emailAddress, validator ) {
-		if ( ! validator ) {
-			validator = () => true;
-		}
+	async pollEmailsByRecipient( emailAddress, validator = () => true ) {
 		const intervalMS = 1500;
 		let retries = emailWaitMS / intervalMS;
 		let emails;
 
 		while ( retries > 0 ) {
-			retries--;
 			emails = await this.mailbox.getEmailsByRecipient( emailAddress );
 			if ( validator( emails ) ) {
 				return emails;
 			}
+			await this.resolveAfterTimeout( intervalMS );
+			retries--;
 		}
+		throw new Error( `Could not locate email for '${ emailAddress }' in '${ emailWaitMS }'ms` );
+	}
+
+	resolveAfterTimeout( timeout ) {
```

Comment:
I put this instead of `setInterval()` and it should work with disabled promise manager.